### ISSUE TITLE
RNN validation loss NaN and crashes EarlyStopping, but normal loss is fine

### ISSUE THREAD

**Issue by jneeven** [AppliedDataSciencePartners/WorldModels#14]
If I train the RNN with default configuration, the validation losses computed at the end of each epoch are NaN. This is strange, because the normal loss computed every batch seems to work fine. 
Since the EarlyStopping callback monitors validation loss, the entire training script crashes after 5 epochs (since the tolerance is 5 and the callback therefore starts comparing NaN values after the 5th epoch)...

Any ideas what might cause this?

**Comment by jneeven** [comment]
It turns out that some entries in my observation data were corrupt, looking very different from the other images. Since the VAE isn't familiar with such images, it predicts a very large or tiny mean and log variance, which in turn leads to the standard deviation becoming infinite. 

This infinite data of course results in NaN predictions when fed into the RNN.
See attached picture for illustration:
![corrupt_images batch9_189_-75](https://user-images.githubusercontent.com/19648598/52724885-77131680-2fb0-11e9-933f-3d7f3ee0b34d.png)

I will close this issue since I don't think file corruption is relevant to this repo in particular, although it might be wise to build some kind of check for this if you plan to replicate the world models results (if the encoding is infinite at any point, I simply ignore the entire episode in generate_rnn_data).
